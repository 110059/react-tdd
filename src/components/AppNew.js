import React, { useState, useEffect, useReducer } from 'react';
import axios from 'axios';

export const dataReducer = (state, action) => {
  if (action.type === 'SET_ERROR') {
    return { ...state, list: [], error: true };
  }
  if (action.type === 'SET_LIST') {
    return { ...state, list: action.list, error: null };
  }
  throw new Error();
};

const initialData = {
  list: [],
  error: null,
};

const AppNew = () => {
  const [counter, setCounter] = useState(0);
  const [data, dispatch] = useReducer(dataReducer, initialData);

    useEffect(() => {
    axios
      .get('http://hn.algolia.com/api/v1/search?query=react')
      .then(response => {
        dispatch({ type: 'SET_LIST', list: response.data.hits });
      })
      .catch(() => {
        dispatch({ type: 'SET_ERROR' });
      });
  }, []);

  return (
    <div>
      <h1>My Counter</h1>
      <Counter counter={counter} />
      <button type="button" onClick={() => setCounter(counter + 1)}>
        Increment
      </button>
      <button type="button" onClick={() => setCounter(counter - 1)}>
        Decrement
      </button>
      <h2>My Async Data</h2>
      {data.error && <div className="error">Error</div>}
      <ul>
        {data.list.map(item => (
          <li key={item.objectID}>{item.title}</li>
        ))}
      </ul>
    </div>
  );
};

export const Counter = ({ counter }) => (
  <div>
    <p>{counter}</p>
  </div>
);

export default AppNew;