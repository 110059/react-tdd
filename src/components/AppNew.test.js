import renderer from 'react-test-renderer';
import React from 'react';
import { Counter } from './AppNew';

describe("Counter", () => {
    test('snap shot render', ()=> {
        const component = renderer.create(<Counter counter="1" />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
