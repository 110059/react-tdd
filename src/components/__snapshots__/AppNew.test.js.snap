// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snap shot render 1`] = `
<div>
  <p>
    1
  </p>
</div>
`;
